app/assets/javascript/application.js

```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import App from './components/App'
+
+
+document.addEventListener("DOMContentLoaded", () => {
+    ReactDOM.render(<App />, document.getElementById('root'));
+  });
```

Comment:
- [ ] I can see that your application folder is inside the asset folder, it shouldn't be a sister folder to the asset folder and not a child, I will point the javascript folder it should be.